### PR TITLE
Remove useless codes about minion

### PR DESCRIPTION
**Pull request by feihujiang**
Minion isn't used now, so remove codes about it.

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra @kubernetes/rh-ux

**Comment by jlowdermilk** [comment]
Pretty sure this code is safe to rip out now, but will let someone else lgtm.

cc @bgrant0607 @lavalamp

File: pkg/api/v1/conversion_test.go [hunk 1]
```diff
@@ -23,28 +23,28 @@ import (
...
 
-func TestNodeConversion(t *testing.T) {
-	obj, err := versioned.Codec.Decode([]byte(`{"kind":"Minion","apiVersion":"v1"}`))
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	if _, ok := obj.(*api.Node); !ok {
-		t.Errorf("unexpected type: %#v", obj)
-	}
+//func TestNodeConversion(t *testing.T) {
```
Comment (lavalamp) [inline pkg/api/v1/conversion_test.go]: Please delete instead of commenting out. (git keeps a record should we ever need it again)

**Comment by lavalamp** [comment]
This is fine once tests pass.

**Comment by feihujiang** [comment]
I can't access Shippable Details, who can post them for me, thanks very much!

**Comment by jlowdermilk** [comment]
Test failure is timeout, unrelated to this pr. Kicked and will look into timeouts separately.

**Comment by jlowdermilk** [comment]
Actually, it looks like there is a test referencing the minion resource:
```
--- FAIL: TestResourceTuple (0.01s)
	builder_test.go:635: valid multiple with namespaced and non-namespaced types: unexpected error: no resource "minions" has been defined
```

**Comment by feihujiang** [comment]
@jlowdermilk Thanks for posting the failure

**Comment by feihujiang** [comment]
@lavalamp I updated this commit according to your advice, please review.

**Comment by lavalamp** [comment]
LGTM, thanks

**Pull request merged**